src/lib/api/org/index.ts

```diff
@@ -275,3 +277,108 @@ function convertToSnykProject(projectData: v3ProjectData[]) : SnykProject[] {
   return projects
 }
 
+export async function listTargets(
+  requestManager: requestsManager,
+  orgId: string,
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }
+): Promise<TargetsResponse> {
+  getApiToken();
+  getSnykHost();
+  debug(`Listing all targets for org: ${orgId}`);
+
+  if (!orgId) {
+    throw new Error(
+      `Missing required parameters. Please ensure you have set: orgId, settings.
+        \nFor more information see: https://apidocs.snyk.io/?version=2022-09-15%7Ebeta#get-/orgs/-org_id-/targets`,
+    );
+  }
+  
+  const targets = await listAllSnykTarget(requestManager, orgId, config)
+
+  return {targets: targets};
+}
+
+export async function listAllSnykTarget(requestManager: requestsManager,
+  orgId: string,
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }
+  ): Promise<SnykTarget[]> {
+
+    let lastPage = false;
+    let SnykTargetList: SnykTarget[] = [];
+    let nextPageLink: string | undefined = undefined;
+    while (!lastPage) {
+      try {
+        const {
+          SnykTarget,
+          next,
+        }: {
+          SnykTarget: SnykTarget[];
+          next?: string;
+        } = await getSnykTarget(requestManager, orgId, nextPageLink, config);
+
+        SnykTargetList = SnykTargetList.concat(SnykTarget);
+        next
+          ? ((lastPage = false), (nextPageLink = next))
+          : ((lastPage = true), (nextPageLink = ''));
+      } catch (e) {
+        debug('Failed to get targets for ', orgId, e);
+        throw e;
+    }
+  }
+  return SnykTargetList
+}
+
+export async function getSnykTarget(requestManager: requestsManager,
+  orgId: string,
+  nextPageLink?: string, 
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }): Promise< { SnykTarget: SnykTarget[], next?: string } > {
+
+    const link = nextPageLink ? nextPageLink : `https://api.dev.snyk.io/rest/orgs/${orgId.trim()}/targets?version=2022-09-15~beta`
+    const url = new URL(link)
+
+    if (config) {
```

Comment:
If all entries in "config" are there, we would go through all the "if" statements, was that the intention or should we apply "else" here?

Reply:
config can be undefined, so we need to check and same for the option on config

Reply:
no need for else we apply this only if config exist